logzio-sender/src/main/java/io/logz/sender/InMemoryQueue.java

```diff
@@ -1,42 +1,44 @@
 package io.logz.sender;
 
-import io.logz.sender.exceptions.LogzioParameterErrorException;
-
 import java.io.IOException;
 import java.util.concurrent.ConcurrentLinkedQueue;
-import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.ReentrantLock;
 
-public class InMemoryLogsBuffer implements LogzioLogsBufferInterface {
-    private static final int ONE_MEGA_BYTE_IN_BYTES = 1024 * 1024;
+public class InMemoryQueue implements LogsQueue {
+    private static final int MG_IN_BYTES = 1024 * 1024;
     public static int DONT_LIMIT_BUFFER_SPACE = -1;
     private final ConcurrentLinkedQueue<byte[]> logsBuffer;
     private final boolean dontCheckEnoughMemorySpace;
-    private final int bufferThreshold;
+    private final int capacityInBytes;
     private final SenderStatusReporter reporter;
-    private AtomicInteger size;
-
-    private InMemoryLogsBuffer(boolean dontCheckEnoughMemorySpace, int bufferThreshold, SenderStatusReporter reporter)
-            throws LogzioParameterErrorException {
+    private int size;
+    private ReentrantLock queueLock;
 
+    private InMemoryQueue(boolean dontCheckEnoughMemorySpace, int capacityInBytes, SenderStatusReporter reporter) {
         logsBuffer = new ConcurrentLinkedQueue<>();
         this.dontCheckEnoughMemorySpace = dontCheckEnoughMemorySpace;
-        this.bufferThreshold = bufferThreshold;
+        this.capacityInBytes = capacityInBytes;
         this.reporter = reporter;
-        this.size = new AtomicInteger();
+        this.size = 0;
+        this.queueLock = new ReentrantLock();
     }
 
     @Override
     public void enqueue(byte[] log) {
+        queueLock.lock();
```

Comment:
Please read the javadoc before using. 
After lock() you should
try {
...
} finally {
   queueLock.unlock();
}

This protects against exception that leave lock always locked.
Javadoc has recipe on using.